packages/model-generator/src/create_graphql_document_generator.ts

```diff
@@ -37,7 +40,25 @@ export const createGraphqlDocumentGenerator = ({
   const fetchSchema = async () => {
     const backendOutputClient =
       BackendOutputClientFactory.getInstance(awsClientProvider);
-    const output = await backendOutputClient.getOutput(backendIdentifier);
+    let output;
+    try {
```

Comment:
rather than having these try/catch blocks in the model-generator, should we instead move these to where the model-gen package is used by the CLI? Similar to the changes you made to the form-gen command above.

Reply:
I'm adding the try/catches at the call site of `BackendOutputClient.getOutput()` instead of wrapping the consumers everywhere. The drawback of that is if these "Generators" are used elsewhere, we would forget to wrap these errors again.

I'd have preferred to add a single try/catch in the `getOutput` implementation, but that is shared between console and CLI so couldn't do it.

Reply:
Maybe we could have a function internal to the model-gen package like `getOutputWithErrorHandling` or something so we don't have to repeat the same error handling logic in multiple places? Or if we really want to go nuts with the DI pattern, have a `new ErrorHandlingBackendOutputClient(backendOutputClient)` that we use internally to this package.

I'm open to whatever pattern you like, I'm just concerned about having the same error handling logic in multiple places.

Reply:
Updated to use a common handler logic.